runtime/string_functions.cpp

```diff
@@ -2011,8 +2009,8 @@ string f$strtoupper(const string &str) {
 
   // same optimization as in strtolower
   const char *end = str.c_str() + n;
-  const char *lowercase_pos = std::find_if(str.c_str(), end, [](unsigned char ch) {
-    return ch >= 'a' && ch <= 'z';
+  const char *lowercase_pos = std::find_if(str.c_str(), end, [](char ch) {
+    return php_islower(ch);
   });
```

Comment:
The lambda is redundant here
```suggestion
  const char *lowercase_pos = std::find_if(str.c_str(), end, php_islower);
```